### ISSUE TITLE
Error TS2304 when I try to compile with Typescript 

### ISSUE THREAD

**Issue by lorflea** [amz-tools/amazon-sp-api#99]
Hi,
When I try to compile my project that includes amazon-sp-api, I get the following Typescript error:

```
node_modules/amazon-sp-api/lib/typings/index.d.ts(436,7): error TS2304: Cannot find name 'CreateFeedBody'.
node_modules/amazon-sp-api/lib/typings/index.d.ts(446,7): error TS2304: Cannot find name 'CreateReportBody'.
node_modules/amazon-sp-api/lib/typings/index.d.ts(453,21): error TS2304: Cannot find name 'IReqOptions'.
node_modules/amazon-sp-api/lib/typings/index.d.ts(463,3): error TS2666: Exports and export assignments are not permitted in module augmentations.
```

**Comment by amz-tools** [comment]
Hi @lorflea,

sorry for that, is it possible for you to fix this in a PR?

**Comment by lorflea** [comment]
Sure, I will submit a PR once fixed.

Il giorno lun 31 gen 2022 alle 19:47 amz-tools ***@***.***>
ha scritto:

**Comment by amz-tools** [comment]
@lorflea,

have you had the chance to look into it? Otherwise, maybe @omrishaked could help out with a quick fix?

**Comment by omrishaked** [comment]
Just added the latest `amazon-sp-api` version to my Typescript project, imported it (`import SellingPartner from 'amazon-sp-api';`) and even created an instance (`const sellingPartner = new SellingPartner({ region: 'eu' });`), compiled (using `tsc` and `esbuild`) with no issues at all.  Not sure what the problem here, but I also looked at the types file and it looks good to me.

**Comment by PhilipKazmeier** [comment]
Hey, I have the same errors as @lorflea.
@omrishaked could you share the files and tsconfig you used?

**Comment by Tobier1988** [comment]
Hey, 
i applied the following changes using [https://www.npmjs.com/package/patch-package](url) to the lib. This solved the issue for my setup. However i am not sure if this fix is valid for all possible setups. 

[https://gist.github.com/Tobier1988/bc191cb32a689e162166f653df4443d9](url)

**Comment by Tobier1988** [comment]
Just to add to that, I created this stackblitz as a reproduction case:

https://stackblitz.com/edit/node-q2yggy?file=index.ts

Just run `npx tsc` in the command line and it will show 4 errors.

Note that this only happens when skibLibCheck in tsconfig is set to false.

**Comment by TraceM171** [comment]
> Hey, i applied the following changes using [https://www.npmjs.com/package/patch-package](url) to the lib. This solved the issue for my setup. However i am not sure if this fix is valid for all possible setups.
> 
> [https://gist.github.com/Tobier1988/bc191cb32a689e162166f653df4443d9](url)

Any updates in this issue? I tried this fix but an error raises saying that class SellingPartner is not a constructor.

**Comment by amz-tools** [comment]
@TraceM171 I know this is a very old issue but it should be fixed now in version v0.8.4.
